llama-index-integrations/vector_stores/llama-index-vector-stores-elasticsearch/llama_index/vector_stores/elasticsearch/base.py

```diff
@@ -531,57 +442,25 @@ async def aquery(
             Exception: If AsyncElasticsearch query fails.
 
         """
-        query_embedding = cast(List[float], query.query_embedding)
-
-        es_query = {}
+        _mode_must_match_retrieval_strategy(query.mode, self.retrieval_strategy)
 
         if query.filters is not None and len(query.filters.legacy_filters()) > 0:
```

Comment:
fyi this is why less filters are supported, elastic is only using the legacy exact match filters. If updated, it could use many filters (less than, greater than, contains, etc.) and operators (AND, OR)